### PR TITLE
feat(react-native): add convert-to-inferred generator for Expo and React Native

### PR DESCRIPTION
**Pull request by jaysoo**
This PR adds `convert-to-inferred` generators to convert React Native an Expo apps using executors to use the corresponding inference plugins.

Also:
1. Fixes casing for `@nx/react-native/plugin` so it is correctly set as `upgradeTargetName` not `upgradeTargetname`
2. Migration for the above fix for existing projects

### CODE REVIEW

File: packages/expo/src/executors/run/schema.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     },
     "port": {
       "type": "number",
-      "description": "Port to start the Metro bundler on",
+      "description": "Port to start the Metro Gbundler on",
```
Comment (ndcunningham) [inline packages/expo/src/executors/run/schema.json:40]: typo?

Comment (ndcunningham) [inline packages/expo/src/executors/run/schema.json:40]: typo?

File: packages/expo/src/generators/convert-to-inferred/lib/post-target-transformer.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { type TargetConfiguration, type Tree } from '@nx/devkit';
```
Comment (ndcunningham) [inline packages/expo/src/generators/convert-to-inferred/lib/post-target-transformer.ts:1]:
```suggestion
import type { TargetConfiguration, Tree } from '@nx/devkit';
```

File: packages/expo/src/generators/convert-to-inferred/lib/process-prebuild-options.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { Tree } from '@nx/devkit';
```
Comment (ndcunningham) [inline packages/expo/src/generators/convert-to-inferred/lib/process-prebuild-options.ts:1]:
```suggestion
import type { Tree } from '@nx/devkit';
```

File: packages/expo/src/generators/convert-to-inferred/lib/process-install-options.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { Tree } from '@nx/devkit';
```
Comment (ndcunningham) [inline packages/expo/src/generators/convert-to-inferred/lib/process-install-options.ts:1]:
```suggestion
import type { Tree } from '@nx/devkit';
```

File: packages/expo/src/generators/convert-to-inferred/lib/process-build-options.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { Tree } from '@nx/devkit';
```
Comment (ndcunningham) [inline packages/expo/src/generators/convert-to-inferred/lib/process-build-options.ts:1]:
```suggestion
import type { Tree } from '@nx/devkit';
```

**Review by isaacplmann — APPROVED** [review]

**Review by Coly010 — APPROVED** [review]
1 comment, LGTM!

File: packages/react-native/src/migrations/update-19-6-0/rename-upgrade-target-name.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  Tree,
+  formatFiles,
+  getProjects,
+  readNxJson,
+  updateNxJson,
+  updateProjectConfiguration,
+  workspaceRoot,
+} from '@nx/devkit';
+
+import { runCliUpgrade } from '../../executors/upgrade/upgrade.impl';
```
Comment (Coly010) [inline packages/react-native/src/migrations/update-19-6-0/rename-upgrade-target-name.ts:11]: Is this being used?

**Pull request merged**